### PR TITLE
Make a service of QRGenerator

### PR DESCRIPTION
**Pull request by jverbraeken**
- Relevant Issues: -
- Related Pull Requests: -

* * *

## What
This Pull Request adds to the repository the transfer of QRGenerator to the newly created util service

## Why
This Pull Request is needed because classes that offer a platform-dependent service should be encapsulated in a service to reduce the coupling between the application and the platform

## How
This feature can be viewed/tested within the project by checking if things are still working; running QRGeneratorTest

## Alternative implementation
-

### CODE REVIEW

**Comment by jverbraeken** [comment]
> This will most certainly lead to complications with PR #136... Maybe we should merge it into this branch to make sure to avoid having to resolve these later? Is it OK if I merge that branch into this PR?

Oops, didn't see that. I'll fix the conflicts...

File: app/src/androidTest/java/com/nervousfish/nervousfish/test/BaseTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+/**
+ * Provides Powermock-functionality to tests.
+ */
+public final class BaseTest {
```
Comment (clenrock) [inline app/src/androidTest/java/com/nervousfish/nervousfish/test/BaseTest.java]: What is this BaseTest for? Is it used?

Comment (jverbraeken) [inline app/src/androidTest/java/com/nervousfish/nervousfish/test/BaseTest.java]: Yes, it's used by QRGeneratorTest to instantiate a private object

File: app/src/androidTest/assets/features/bluetoothConnection.feature [hunk 2]
```diff
@@ -0,0 +1,20 @@
+Feature: Bluetooth connection service
```
Comment (ericcornelissen) [inline app/src/androidTest/assets/features/bluetoothConnection.feature]: This seems like the wrong PR for this file...?

File: app/src/main/java/com/nervousfish/nervousfish/modules/util/AUtils.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package com.nervousfish.nervousfish.modules.util;
+
+/**
+ * This class should be extended by all platform-specific {@link IUtils} implementations and contains
+ * all cross-platform functionality
+ */
+@SuppressWarnings("PMD.AbstractClassWithoutAnyMethod")
+abstract class AUtils implements IUtils {
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/modules/util/AUtils.java]: An abstract utilities class that is focused on QR? That seems a bit weird to me...?

File: app/src/main/java/com/nervousfish/nervousfish/modules/util/AndroidUtils.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+public final class AndroidUtils extends AUtils {
+    private static final Logger LOGGER = LoggerFactory.getLogger("EncryptorAdapter");
+
+    /**
+     * Prevents construction from outside the class.
+     *
+     * @param serviceLocator Can be used to get access to other modules
+     */
+    @SuppressWarnings("PMD.UnusedFormalParameter")
+    // This servicelocator will be used later on probably
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/modules/util/AndroidUtils.java]: The `serviceLocator` is being used, right?

File: app/src/main/java/com/nervousfish/nervousfish/service_locator/ServiceLocator.java [hunk 5]
```diff
@@ -16,13 +16,15 @@
...
+import com.nervousfish.nervousfish.modules.util.AndroidUtils;
+import com.nervousfish.nervousfish.modules.util.IUtils;
 
 import org.greenrobot.eventbus.EventBus;
 
 /**
  * Manages all modules and provides access to them.
  */
-final class ServiceLocator implements IServiceLocator {
+public final class ServiceLocator implements IServiceLocator {
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/service_locator/ServiceLocator.java]: Why is it `public` now?

**Comment by jverbraeken** [comment]
Oops, I think that something went wrong when I made this branch. There are far too many files changed for such a simple PR.

**Comment by jverbraeken** [comment]
I'll make a new branch and make a new PR for that one

**Pull request closed without merging**